test/fetch_test.rb

```diff
@@ -342,6 +342,68 @@ def test_respects_should_use_cache_on_record
     end
   end
 
+  def test_with_should_use_cache_true_fetched_record_does_not_have_embedded_associations_preloaded_on_cache_miss
+    Item.cache_has_many(:associated_records, embed: true)
+    Item.cache_has_many(:normalized_associated_records)
+
+    @record.associated_records.build
+    @record.associated_records.build
+    @record.normalized_associated_records.build
+    @record.save
+
+    assert_memcache_operations(1) do
+      assert_queries(3) do
+        item = Item.fetch_by_id(@record.id)
+
+        refute(item.association(:associated_records).loaded?)
+        refute(item.association(:normalized_associated_records).loaded?)
+      end
+    end
+  end
+
+  def test_with_should_use_cache_true_fetched_record_does_not_have_embedded_associations_preloaded_on_cache_hit
+    Item.cache_has_many(:associated_records, embed: true)
+    Item.cache_has_many(:normalized_associated_records)
+
+    @record.associated_records.build
+    @record.associated_records.build
+    @record.normalized_associated_records.build
+    @record.save
+
+    # warm the cache
+    Item.fetch_by_id(@record.id)
+
+    assert_memcache_operations(1) do
+      assert_queries(0) do
+        item = Item.fetch_by_id(@record.id)
+
+        refute(item.association(:associated_records).loaded?)
+        refute(item.association(:normalized_associated_records).loaded?)
+      end
+    end
+  end
+
+  def test_with_should_use_cache_false_fetched_record_has_embedded_associations_preloaded
+    Item.cache_has_many(:associated_records, embed: true)
+    Item.cache_has_many(:normalized_associated_records)
+
+    @record.associated_records.build
+    @record.associated_records.build
+    @record.normalized_associated_records.build
+    @record.save
+
+    Item.stubs(:should_use_cache?).returns(false)
+
+    assert_memcache_operations(0) do
+      assert_queries(2) do
+        item = Item.fetch_by_id(@record.id)
+
+        assert(item.association(:associated_records).loaded?)
+        refute(item.association(:normalized_associated_records).loaded?)
+      end
```

Comment:
Avoid making these blocks larger than they need to be, so the test is clearer (e.g. don't need to read past that to see if the assertion lines are relevant to the query count)

```suggestion
      item = assert_queries(2) do
        Item.fetch_by_id(@record.id)
      end

      assert(item.association(:associated_records).loaded?)
      refute(item.association(:normalized_associated_records).loaded?)
```